tree/30_generic_methods/_logger.cf

```diff
@@ -19,20 +19,22 @@
 # @name Logger
 # @description Standard logging output
 #
-# @parameter message      The common part of the message to display
-# @parameter class_prefix The prefix of the class for different states
+# @parameter message          The common part of the message to display
+# @parameter old_class_prefix The prefix of the class for different states
 #
 # @class_prefix
 # @class_parameter
 # The three states are kept, repaired and not_ok
 # (as defined in the classes_generic of the cfengine_stdlib)
 
-bundle agent _logger(message, class_prefix)
-{ 
+bundle agent _logger(message, old_class_prefix, args)
+{
+  vars:
+      "method" string => "${configuration.enabled_loggers}";
 
   methods:
-
       # Report using the appropriate bundle(s)
-      "logging" usebundle => _bundle_caller_two("${configuration.enabled_loggers}", "${message}" , "${class_prefix}");
-
+      "${method}"
+        usebundle => ${method}("${message}", "${old_class_prefix}", "${args}"),
```

Comment:
can't you directly ${configuration.enabled_loggers}("${message}", "${old_class_prefix}", "${args}") ?